runtimes/alibi-detect/mlserver_alibi_detect/runtime.py

```diff
@@ -105,23 +118,37 @@ def _detect(self, payload: InferenceRequest) -> InferenceResponse:
         input_data = self.decode_request(payload, default_codec=NumpyRequestCodec)
         predict_kwargs = self._ad_settings.predict_parameters
 
-        try:
-            y = self._model.predict(np.array(input_data), **predict_kwargs)
-            return self._encode_response(y)
-        except (ValueError, IndexError) as e:
-            raise InferenceError(
-                f"Invalid predict parameters for model {self._settings.name}: {e}"
-            ) from e
+        # If batch is configured, wrap X in a list so that it is not unpacked
+        X = np.array(input_data)
+        if self._ad_settings.batch_size:
+            X = [X]
```

Comment:
Wrap `X` in a list (of length 1) so that we don't unpack into single instances when running in batch mode (i.e. offline detectors). `pred` is then returned as a list of length 1. 

This slightly wacky logic is intended to avoid having two duplicate try-except blocks for the two (offline vs online) use cases...